code/game/objects/effects/decals/remains.dm

```diff
@@ -49,6 +54,28 @@
 	cigarette_puff.set_up(range = 2, amount = DIAMOND_AREA(2), holder = src, location = get_turf(src), silent = TRUE)
 	cigarette_puff.start()
 
+///Returns a random reagent object minus blacklisted reagents and with other blocking conditions to make the potential choices more interesting.
+/obj/effect/decal/remains/human/smokey/proc/generate_reagent()
+	var/static/list/random_reagents = list()
+	if(!length(random_reagents))
+		for(var/datum/reagent/reagent_path as anything in subtypesof(/datum/reagent))
+			if(!(initial(reagent_path.chemical_flags) & REAGENT_CAN_BE_SYNTHESIZED))
+				continue
+			if(istype(reagent_path, /datum/reagent/medicine) || istype(reagent_path, /datum/reagent/consumable)) //Boooooriiiiing
+				continue
+			random_reagents += reagent_path
+	var/picked_reagent = pick(random_reagents)
+	return picked_reagent
```

Comment:
Please use get_random_reagent_id() instead. You can make it respect a blacklist to get the same effect (or flags or something, but blacklist is probably easiest)